pkg/kubelet/dockershim/docker_service.go

```diff
@@ -397,11 +402,17 @@ func (ds *dockerService) GetPodPortMappings(podSandboxID string) ([]*hostport.Po
 // Start initializes and starts components in dockerService.
 func (ds *dockerService) Start() error {
 	// Initialize the legacy cleanup flag.
+	if ds.startLocalStreamingServer {
+		go func() {
+			if err := ds.streamingServer.Start(true); err != nil {
+				glog.Errorf("Failed to start streaming server: %v", err)
```

Comment:
should this fatal kubelet instead of only logging a error message

Reply:
Just fixed a similar issue in containerd, will do.

Reply:
> Can we block until the server has started, and return error if it fails?

Nope, `Start` is a blocking call.

Reply:
Done.